### PR TITLE
tests(core): Introduce entity mocks into deck

### PR DESCRIPTION
**Pull request by caseyhebebrand**
Introduce some common entity mocks and use them in a sample test. Please read [this doc](https://docs.google.com/document/d/1pAbx1esmGUvyr1Dm8oeLQ_zL9snVjZG8P1H1eAoewkY/edit?usp=sharing) to learn more about my vision for Deck's testing story.

### CODE REVIEW

File: app/scripts/modules/core/src/mocks/mockLoadBalancer.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { mockInstance, mockInstanceCounts, mockMoniker } from 'core/mocks';
+
+export const mockLoadBalancer: ILoadBalancer = {
+  account: 'test',
+  cloudProvider: 'aws',
+  detail: 'testing',
+  healthState: 'Up',
+  instanceCounts: mockInstanceCounts,
+  instances: [mockInstance],
+  loadBalancerType: 'applciation',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/mocks/mockLoadBalancer.ts]:
```suggestion
  loadBalancerType: 'application',
```

File: app/scripts/modules/core/src/mocks/mockSubnet.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  availabilityZone: 'us-west-1b',
+  id: 'subnet-f7af8783',
+  name: 'test-subnet',
+  account: 'test',
+  region: 'us-west-1',
+  type: 'aws',
+  label: 'label',
+  purpose: 'testing subnets',
+  deprecated: false,
+  vpcId: 'vpc-9af769ff',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/mocks/mockSubnet.ts]:
```suggestion
  vpcId: 'vpc-1234',
```

File: app/scripts/modules/core/src/mocks/mockSubnet.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { ISubnet } from '../domain/ISubnet';
+
+export const mockSubnet: ISubnet = {
+  availabilityZone: 'us-west-1b',
+  id: 'subnet-f7af8783',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/mocks/mockSubnet.ts]:
```suggestion
  id: 'subnet-1234',
```

File: app/scripts/modules/core/src/mocks/mockInstance.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { IInstance } from 'core/domain';
+import { mockHealth } from 'core/mocks';
+
+export const mockInstance: IInstance = {
+  account: 'test',
+  availabilityZone: 'us-west-2',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/mocks/mockInstance.ts]:
```suggestion
  availabilityZone: 'us-west-2a',
```

File: app/scripts/modules/core/src/index.ts [hunk 5]
```diff
@@ -48,6 +48,8 @@ export * from './managed';
 
 export * from './manifest';
 
+export * from './mocks';
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/index.ts]: I'm not sure we want the mocks exported from (and bundled into) `@spinnaker/core`.  @caseyhebebrand what do you think?

**Comment by christopherthielen** [comment]
This is cool.  Is your plan to use these going forward, or to retrofit some other tests with the common mocks?

File: app/scripts/modules/core/src/index.ts (hunk 5 above)
Comment (caseyhebebrand) [reply app/scripts/modules/core/src/index.ts]: This was one thing I wanted thoughts on. We could either:
1. Keep mocks tied closely with the `domain` folder of a Spinnaker package, therefore living in various packages. 
2. Create one `mocks` subdirectory that references types from all areas of Spinnaker. 

I am more in favor of #2, so tests do not have to be bundled to `spinnaker/core` if they don't have to be. I just wasn't sure if people would be in favor of relying on another package.

**Comment by caseyhebebrand** [comment]
> This is cool. Is your plan to use these going forward, or to retrofit some other tests with the common mocks?

A little of both. Its easy enough to use it going forward and when I have some time I will retrofit old tests.

File: app/scripts/modules/core/src/index.ts (hunk 5 above)
Comment (christopherthielen) [reply app/scripts/modules/core/src/index.ts]: I don't know what my opinion is for 1 vs 2

my question is more about the re-export here.  By exporting the mocks at the top level `index.ts`, it guarantees that the mock code is included in the production core bundle.  However, we only want to use mocks in tests.  

I think this export should be removed from `index.ts`.  Tests will still need access, but I think they can `import { fooMock } from '@spinnaker/core/mocks'`

**Pull request merged**